app/models/talking.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class Talking < ApplicationRecord
+  belongs_to :user
+
+  validates :event_name, :talk_title, :date, presence: true
+  validate :future_date?
+
+  private
+
+  def future_date?
+    return if date.blank?
+
+    errors.add(:date, "must be before the today's date") if date.future?
```

Comment:
Add the translation in the i18n here; active record already provides an approach for this